ios/app/brave_main_delegate.mm

```diff
@@ -0,0 +1,68 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/app/brave_main_delegate.h"
+
+#include "base/command_line.h"
+#include "base/files/file_path.h"
+#include "base/mac/bundle_locations.h"
+#include "base/base_paths.h"
+#include "base/path_service.h"
+#include "components/sync/driver/sync_driver_switches.h"
+#include "components/sync/base/model_type.h"
+#include "components/browser_sync/browser_sync_switches.h"
+#include "components/sync/base/sync_base_switches.h"
+#include "ios/chrome/browser/chrome_switches.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+namespace {
+
+#if defined(OFFICIAL_BUILD)
```

Comment:
these should go in gn so they aren't duplicated cc @darkdh

Reply:
yes, we can put in into `defines` so that it can be shared among different files and config it through gn
Just like https://github.com/brave/brave-core/commit/df0a12dd9115a0ad4252f0d7eeccabf1efb57bea#diff-79ce6e05ced60a2fae4287c464e43fe960754823a51326886954af66cd27c775

Reply:
Or just through `public_config`

Reply:
https://github.com/brave/brave-core/pull/7067

Reply:
PR has been merged, please rebase and access the sync url from gn